### PR TITLE
Fix Stylelint errors for no-duplicate-selectors

### PR DESCRIPTION
**Pull request by manzoorwanijk**
#### Proposed Changes

See #67467 

- Fix Stylelint errors for `no-duplicate-selectors` rule.
- In case of conflict, the declarations with `!important` flag and the declarations defined later have been retained.

#### Testing Instructions

- Do a sanity check on Calypso live link

### CODE REVIEW

File: client/blocks/conversation-follow-button/style.scss [hunk 1]
```diff
@@ -72,13 +74,13 @@ button.conversation-follow-button {
...
 		&:focus {
-			color: var( --color-success );
+			color: var(--color-success);
 
 			.reader-following-conversation {
-				fill: var( --color-success );
+				fill: var(--color-success);
 				g {
 					fill: white;
-					stroke: var( --color-success );
```
Comment (noahtallen) [inline client/blocks/conversation-follow-button/style.scss:81]: Are there any disable comments in the file which prevented these from being auto-fixed? (or maybe it's a shortcoming of the autofixer)

**Review by noahtallen — CHANGES_REQUESTED** [review]
Very nice! This is one of the best improvements yet just because duplicated styles make it difficult for devs to understand how the code works.

Given the extent of the changes, I think this PR is moderately risky, so we should be somewhat cautious. I did some smoke testing and found two issues (differences between calypso.live and trunk) that would be worth double checking:

### My sites search bar has odd style behind the "x" button:
SCR-20221007-e6r

### The "write" button in the top bar is misaligned and also missing the chevron "down" button:

https://user-images.githubusercontent.com/6265975/194506834-feb93382-9e2f-4b92-9c43-459ab90239e2.mov

File: client/blocks/conversation-follow-button/style.scss (hunk 1 above)
Comment (manzoorwanijk) [reply client/blocks/conversation-follow-button/style.scss:81]: No, the changes are from a recent PR where the dev doesn't seem to have the IDE properly set up and it looks like `--no-verify` flag was passed when committing the changes.

Comment (noahtallen) [reply client/blocks/conversation-follow-button/style.scss:81]: Got it, thanks!

**Comment by manzoorwanijk** [comment]
> Given the extent of the changes, I think this PR is moderately risky, so we should be somewhat cautious.

I agree.



> My sites search bar has odd style behind the "x" button:

It's the same in production
Screenshot 2022-10-07 at 12 40 15 PM





> The "write" button in the top bar is misaligned and also missing the chevron "down" button:

Fixed. Thanks

File: client/layout/sidebar/style.scss [hunk 2]
```diff
@@ -199,6 +199,7 @@
 .sidebar__menu.is-togglable .sidebar__menu-link {
 	line-height: 1.5;
 	padding: 7px 16px 7px 20px;
+	padding-left: 55px;
 }
```
Comment (noahtallen) [inline client/layout/sidebar/style.scss:203]: IMO it'd be slightly better to move these into the `sidebar__menu.is-togglable` section below, just to keep all those styles in the same place

**Review by noahtallen — APPROVED** [review]
Spent some more time smoke testing, and this LGTM 👍

**Comment by noahtallen** [comment]
> It's the same in production

I also noticed this seems to be fixed now in prod and in this branch, so that's good 😁

File: client/layout/sidebar/style.scss (hunk 2 above)
Comment (manzoorwanijk) [reply client/layout/sidebar/style.scss:203]: I agree but I wanted to only fix the lining errors and leave further improvements for future PRs.

**Pull request merged**